challenge.rb

```diff
@@ -1,26 +1,32 @@
 
 def capitalize_each_string(input)
   #implement your solution here
+	return input.map { |string| string.capitalize }
 end
 
 def fetch_the_dog(input)
   #implement your solution here
+	return input.select { |string| string.eql? "dog" }
 end
 
 def no_dogs_allowed(input)
   #implement your solution here
+	return input.reject { |string| string.eql? "dog" }
 end
 
 def count_the_animals(input)
   #implement your solution here
+	return input.count
 end
 
 def fetch_the_first_two(input)
   #implement your solution here
+	return input.take(2)
 end
 
 def fetch_CD_animals(input)
   #implement your solution here
+	return input.find_all { |string| string[0] == 'c' || string[0] == 'd' }
```

Comment:
Ah, so something like

return input.grep(/^(?i)[cd])

assuming fetch_CD_animals doesn't care whether they are capitalized or not.